docs/src/understand/tedge-agent.md

```diff
@@ -0,0 +1,60 @@
+---
+title: The Agent
+tags: [Concept, MQTT]
+sidebar_position: 5
+---
+
+# Thin-edge Agent
+
+Thin-edge agent is a set of services that implement the device management operations on a device:
+- software management, checking installed packages and updating these packages
+- configuration management, checking and updating configuration files
+- firmware management, updating the firmware of the device
+- device restart
+- remote access to a console on the device from the cloud
+- log file management, retrieving excerpts from log files
+
+:::note
+In the current version of thin-edge, the agent features are not implemented by a single executables,
+but by a set of executables:
+
+- `tedge-agent`
+- `c8y-configuration-plugin`
+- `c8y-firmware-plugin`
+- `c8y-log-plugin`
+- `c8y-remote-access-plugin`
+
+The short-term plan is to re-organize these plugins to move the Cumulocity aspects in the Cumulocity
+and to group the management operations in a single executable. 
+:::
+
+Thin-edge agent acts as a device connector:
+- listening to operation requests published on the MQTT bus
+- delegating the actual operations to the operating system
+- reporting progress of the requests
+
+## Operation MQTT topics
+
+Operation requests are published by the requesters on operation specific topics:
+
+- `tedge/commands/req/${operation-type}/${operation-action}`
+- `${operation-type}/${operation-action}` is the well-known name of the operation request,
+  such as `software/update` or `control/restart`
+- the payloads are operation-specific
+
+The corresponding operation responses are published to associated topics:
+
+- `tedge/commands/res/${operation-type}/${operation-action}`.
+
+Here are the topics used by the device management operations
+
+| Operation          | Request Topic                          | Response Topic                         |
+| ------------------ | -------------------------------------- | -------------------------------------- |
+| Get Software List  | `tedge/commands/req/software/list`     | `tedge/commands/res/software/list`     |
+| Software Update    | `tedge/commands/req/software/update`   | `tedge/commands/res/software/update`   |
+| Get Configuration  | `tedge/commands/req/configuration/get` | `tedge/commands/res/configuration/get` |
+| Set Configuration  | `tedge/commands/req/configuration/set` | `tedge/commands/res/configuration/set` |
```

Comment:
I'm assuming that we're trying to document the upcoming contracts here than the current reality. Otherwise the topic keys are `config_snapshot` and `config_update` respectively, instead of `configuration/get` and `configuration/set`.

[Side note] Interesting to see that you also prefer the split subtopic scheme over a single subtopic scheme for operation targets.

Reply:
> I'm assuming that we're trying to document the upcoming contracts here than the current reality.

I didn't wrote this ;-) This has been extracted from [thin-edge.io-specs](https://github.com/thin-edge/thin-edge.io-specs), as this is the only place where  this were described for the main device.
I should have checked accuracy.

=> Updated, as this doc must describe what is implemented.